.github/workflows/ci.yml

```diff
@@ -144,16 +144,30 @@ jobs:
       - name: Check whether HEAD^ contains HotRod storage relevant changes
         run: echo "GIT_HOTROD_RELEVANT_DIFF=$( git diff --name-only HEAD^ | egrep -ic -e '^model/map-hot-rod|^model/map/|^model/build-processor' )" >> $GITHUB_ENV
 
+      - name: Check whether HotRod storage matrix should be executed
+        if: ${{ !endsWith(matrix.server, '-map-hot-rod') || env.GIT_HOTROD_RELEVANT_DIFF != 0 }}
+        run: echo "CHECK_HOTROD_EXECUTION=true" >> $GITHUB_ENV
+
+      - name: Check whether HEAD^ contains JPA map storage relevant changes
+        run: echo "GIT_MAP_JPA_RELEVANT_DIFF=$( git diff --name-only HEAD^ | egrep -ic -e '^model/map-jpa/|^model/map/|^model/build-processor' )" >> $GITHUB_ENV
+
+      - name: Check whether Map-JPA storage matrix should be executed
+        if: ${{ !endsWith(matrix.server, '-map-jpa') || env.GIT_MAP_JPA_RELEVANT_DIFF != 0 }}
+        run: echo "CHECK_MAP_JPA_EXECUTION=true" >> $GITHUB_ENV
+
+      - name: Check whether those phases should be executed
+        run: echo "SHOULD_BE_EXECUTED=$( test "$CHECK_HOTROD_EXECUTION" = true -a "$CHECK_MAP_JPA_EXECUTION" = true && echo true || echo false )" >> $GITHUB_ENV
+
       - name: Cache Maven packages
-        if: ${{ github.event_name != 'pull_request' || matrix.server != 'undertow-map-hot-rod' || env.GIT_HOTROD_RELEVANT_DIFF != 0 }}
+        if: ${{ github.event_name != 'pull_request' || env.SHOULD_BE_EXECUTED == 'true' }}
         uses: actions/cache@v3
```

Comment:
I had a hard time reading this code with the negations and, to understand the final condition `SHOULD_BE_EXECUTED`.

See below for an attempt to remove that last condition, and use `env.SHOULD_BE_EXECUTED != 'false'` for the tests (would need to be applied for all other steps as well). 

```
      - name: Check whether HEAD^ contains HotRod storage relevant changes
        run: echo "GIT_HOTROD_RELEVANT_DIFF=$( git diff --name-only HEAD^ | egrep -ic -e '^model/map-hot-rod|^model/map/|^model/build-processor' )" >> $GITHUB_ENV

      - name: Check whether HotRod storage matrix should be executed
        if: ${{ endsWith(matrix.server, '-map-hot-rod') && env.GIT_HOTROD_RELEVANT_DIFF == 0 }}
        run: echo "SHOULD_BE_EXECUTED=false" >> $GITHUB_ENV

      - name: Check whether HEAD^ contains JPA map storage relevant changes
        run: echo "GIT_MAP_JPA_RELEVANT_DIFF=$( git diff --name-only HEAD^ | egrep -ic -e '^model/map-jpa/|^model/map/|^model/build-processor' )" >> $GITHUB_ENV

      - name: Check whether Map-JPA storage matrix should be executed
        if: ${{ endsWith(matrix.server, '-map-jpa') && env.GIT_MAP_JPA_RELEVANT_DIFF == 0 }}
        run: echo "SHOULD_BE_EXECUTED=false" >> $GITHUB_ENV

      - name: Cache Maven packages
        if: ${{ github.event_name != 'pull_request' || env.SHOULD_BE_EXECUTED != 'false' }}
        uses: actions/cache@v3     
```

Reply:
@sguilhen @ahus1 Thanks for your comments! Yep, I'd say this approach is more readable. 

I wasn't sure about not defined env var, but it never will be evaluated to the false (even to the string format). It should work. Thanks!